cave_miner/search.py

```diff
@@ -81,7 +90,15 @@ def search_pe(filename, cavesize, _bytes):
     section_offset = section.pointer_to_raw_data
     infos = parse_pe_flags(section.characteristics)
     vaddr = section.virtual_address + base_addr
-    search_cave(section.name, section.body, cavesize, section_offset, vaddr, infos, _bytes)
+
+    '''
+    Calculate difference between VirtualSize and RawSize
+    If the RawSize is greater than VirtualSize the system will fill the difference with Zeros, so this space is an code cave
+    '''
+    body = section.body
+    if section.size_of_raw_data > section.virtual_size:
+        body += "\x00" * (section.size_of_raw_data - section.virtual_size) 
```

Comment:
I understand the logic here, but does every format pad they 'missing' virtual address with 0x00 ?

Reply:
Yes, by design at Microsoft Portable Executable and Common Object File Format Specification  (available at: http://bytepointer.com/download.php?name=pecoff_v6.0.doc) at 5.1. Section Data:  

> The data for each section is located at the file offset given by the PointerToRawData field in the section header, and the size of this data in the file is indicated by the SizeOfRawData field. If the SizeOfRawData is less than the VirtualSize, the remainder is padded with zeros.